PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.azureexplorer/src/com/microsoft/azuretools/azureexplorer/editors/QueueFileEditor.java

```diff
@@ -109,6 +119,78 @@ public void createPartControl(Composite composite) {
         createTablePopUp(composite);
     }
 
+    private FileEditorVirtualNode<EditorPart> createVirtualNode(final String name){
+    	final FileEditorVirtualNode<EditorPart> node = new FileEditorVirtualNode<EditorPart>(this, name);
+    	node.addAction(REFRESH, new NodeActionListener() {
+			@Override
+			protected void actionPerformed(NodeActionEvent e) throws AzureCmdException {
+				fillGrid();			
+			}
+		});
+    	
+    	node.addAction(DEQUEUE, new NodeActionListener() {
+			@Override
+			protected void actionPerformed(NodeActionEvent e) throws AzureCmdException {
+				dequeueFirstMessage();			
+			}
+		});
+    	
+    	node.addAction(ADD, new NodeActionListener() {
+			@Override
+			protected void actionPerformed(NodeActionEvent e) throws AzureCmdException {
+				QueueMessageForm queueMessageForm = new QueueMessageForm(PluginUtil.getParentShell(), storageAccount, queue);
+                queueMessageForm.setOnAddedMessage(new Runnable() {
+                    @Override
+                    public void run() {
+                        fillGrid();
+                    }
+                });
+
+                queueMessageForm.open();				
+			}
+		});
+    	
+    	node.addAction(CLEAR_QUEUE, new NodeActionListener() {
+			@Override
+			protected void actionPerformed(NodeActionEvent e) throws AzureCmdException {
+				boolean optionDialog = DefaultLoader.getUIHelper().showConfirmation(
+                        "Are you sure you want to clear the queue \"" + queue.getName() + "\"?",
+                        "Service explorer",
+                        new String[]{"Yes", "No"},
+                        null);
+
+                if (optionDialog) {
+                    DefaultLoader.getIdeHelper().runInBackground(null, "Clearing queue messages", false, true, "Clearing queue messages", new Runnable() {
+                        public void run() {
+                            /*try {
```

Comment:
Same behavior as the original version where the inner codes were commented. I don't know why but just keep the same.